httpcore5/src/main/java/org/apache/hc/core5/http/protocol/RequestContent.java

```diff
@@ -127,4 +133,32 @@ public void process(final HttpRequest request, final EntityDetails entity, final
         }
     }
 
+    /**
+     * Validates the presence and format of the Content-Type header for an OPTIONS request.
+     *
+     * <p>
+     * According to the HTTP protocol, an {@link Method#OPTIONS} request that contains content must also have
```

Comment:
FWIW, this requirement is not really specific to OPTIONS.

Reply:
@reschke 
To ensure I understand correctly, are you suggesting that the validation for a valid Content-Type header when content is present should be applied to all HTTP methods and not just OPTIONS? If so, are there specific methods you have in mind that should undergo this validation? Could you please specify or guide me to the exact section or sections in the RFC where this requirement is detailed for other HTTP methods?

Reply:
What I'm trying to say is that I see little value in accepting invalid Content-Type fields for other methods.

Reply:
@reschke What you are saying makes no sense. @arturobernalg We cannot truly know what if `content-type` value is valid or not unless we maintain a list of vales we consider valid.